### DISCUSSION TITLE
Prioritize the `.ignore` file over `.gitignore` for the file picker.

### DISCUSSION THREAD

**Discussion by MAST1999** [helix-editor/helix#8237] [General] [answered]
This is useful because sometimes there are files that are local and in the `.gitignore` but I want to open them and take a look at them.

So what I ended up trying was putting them inside the `.ignore` file as a not (`!`) rule. For example, `!.env` and then I can see the `.env` file in the picker. 

But currently the picker prioritizes the `.gitignore` so I can't override them in `.ignore`.

If anyone has any other work around as well, please share :smile:

Thanks!

**Comment by pascalkuthe** [comment]
I don't think we can implement that on our end. The ignore handling is handled by the `ignore` library

**Reply by pascalkuthe** [reply, marked as answer]
it acutally should already work, according to the ignre docs `.ignore` files take precedent over `.gitignore` files. The reason the file is likely still hidden here is that it starts with a colon and hence counts as a hidden file

**Comment by michihupf** [comment]
What brought me here is that I did not realize that I had ignored specific files of a directory and was trying to unignore the entire directory. Maybe sharing this helps someone that experiences something similiar. 

Example:
```
dir/
  file1
```

`.gitignore`
```
dir/file1
```

In this case having `!dir/` of course does not unignore `file1`. To make it work we need to unignore the file specifically: `!dir/file1`.
